R/scregclust.R

```diff
@@ -213,6 +218,18 @@ scregclust <- function(expression,
     start_time <- Sys.time()
   }
 
+  if(!(quick_mode %in% c(TRUE, FALSE))){
+    cli::cli_abort("{.var quick_mode} needs to be TRUE or FALSE")
+  }
+
+  if(quick_mode && !allocate_per_obs){
+    cli::cli_abort("{.var quick_mode} is only available when {.var allocate_per_obs} is TRUE")
+  }
+
+  if(!(is.numeric(percent) && 0 <= percent && percent <= 1)){
+    cli::cli_abort("{.var percent} needs to be numeric in [0, 1]")
+  }
+
```

Comment:
Except for `verbose` all arguments are checked in the order they are supplied, so your new arguments should be checked at the very end. (e.g. you are testing for `allocate_per_obs` even though that argument hasn't been checked yet).

Please check existing tests on how to check properly for different types.

e.g. check `verbose` for how to check a single boolean and check `split1_proportion` how to check for a single scalar between 0 and 1 (modified to include 0 and 1 here).

Why is this important?
Your test crashes for `quick_mode` crashes if `quick_mode = c(TRUE, FALSE)` is supplied. The proper version simply gracefully shuts down and emits an error message.

A style note: Always format `if-else`, `for`, ... as
```r
if (test) {
  ...
} else {
  ...
}
```
(note the spaces)

Reply:
Thank you for the information! I have placed my checks at the end and revised them to follow your own examples.

I have reformatted my code to conform to the coding style, as well as changed spacing according to the "80-width" that you mention below. 

Let me know if I have missed anything stylistically!